code-scanning/codegurureviewer.yml

```diff
@@ -0,0 +1,55 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+#
+# For more information on Amazon CodeGuru Reviewer in general, see
+# https://github.com/aws-actions/codeguru-reviewer
+
+name: CodeGuru Reviewer
+
+on:
+  push:
+    branches: [ $default-branch, $protected-branches ]
+  pull_request:
+    branches: [ $default-branch ]
+  schedule:
+    - cron:  $cron-weekly
+  workflow_dispatch:
+    inputs:
+      logLevel:
+        description: 'Log level'
+        required: true
+        default: 'warning'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+  steps:
+    # Step 1: Checkout the repository and provide your AWS credentials
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: us-west-2  # Region to access CodeGuru
+
+    # Step 2: Add CodeGuru Reviewer Action
+    - name: CodeGuru Reviewer
+      uses: aws-actions/codeguru-reviewer@v1.1
```

Comment:
I'm afraid we need a full SHA here, rather than a tag.

Reply:
If I put a full SHA here, does it mean every time my action has an update even the version keeps same, I also need to raise a pr to update the SHA in this repo?

Or could you please elaborate why we need a full SHA here not a tag?

Reply:
I'm afraid it's a requirement we've settled on from a security perspective. When we are providing templates that use actions which are not under our control then a full SHA better ensures users get the expected version.

This does mean updates to the action will requires a small PR here in order to be reflected in the starter workflows page. This shouldn't be any more burdensome than bumping eg `v1.1` to `v1.2`, but apologies that the extra admin is necessary. (We hope that users will find Dependabot useful for keeping their workflows up to date once in use!)

Do let me know if you have further questions! 🙂